cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -157,6 +161,53 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             ;
     }
 
+    static void copy_array_using_ellipsis(wrapped_type & self, pybind11::array & arr_in)
+    {
+        const auto left_shape = self.shape();
+        const auto * right_shape = arr_in.shape();
+
+        auto throw_error = [=]()
+        {
+            std::ostringstream msg;
+            msg << "Input array from shape(";
+            for (pybind11::ssize_t i = 0; i < arr_in.ndim(); ++i)
+            {
+                msg << right_shape[i];
+                if (i != arr_in.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ") into shape(";
+            for (size_t i = 0; i < self.ndim(); ++i)
+            {
+                msg << left_shape[i];
+                if (i != self.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ")";
+
+            throw std::runtime_error(msg.str());
+        };
+
+        if (self.ndim() != static_cast<size_t>(arr_in.ndim()))
+        {
+            throw_error();
+        }
+
+        for (size_t i = 0; i < left_shape.size(); ++i)
```

Comment:
`SimpleArrayFloat64((2, 3, 4))[...] = np.ones(24, dtype='float64')[:]` should work, but this shape test seems to disallow it.  Could you confirm and fix?

I think we should not check for the shape but should check for total element size and item size.

Please also test for `nghost != 0` (see https://github.com/solvcon/modmesh/blob/3b769a47941f5fb2b726de8bb8fca1e6b3f09ea8/tests/test_buffer.py#L254).  The ellipsis syntax should populate everything including the ghost elements.

The full slice (`[:]`) should also cover everything including ghost.  It should use a separate PR, but when the slice works, we may use `[0:]` to populate only the body (excluding ghost).

Note that the ghost index only works in the leading index and the negativity works differently from numpy negative index.

Reply:
The following numpy example gets `ValueError: could not broadcast input array from shape (24,) into shape (2,4,3)`. If we want to follow the behavior of numpy, we should only check the shape.
```
ndarr2 = np.arange(2*3*4, dtype='float64').reshape((2, 4, 3))
ndarr4 = np.ones(24, dtype='float64')[:]
ndarr2[...] = ndarr4[...]
```

Reply:
Could you explain what is a ghost? is there any reference?

Reply:
It's good to follow numpy broadcasting rule.  Let's continue with the shape check.

The test case I posted contains the working example of the ghost.  Here I repost for clarity:

https://github.com/solvcon/modmesh/blob/3b769a47941f5fb2b726de8bb8fca1e6b3f09ea8/tests/test_buffer.py#L254

Reply:
> The ellipsis syntax should populate everything including the ghost elements.
then why do we need to test `nghost != 0`? `memcpy` will have also copy ghost elements.

current PR behavior
```
sarr = modmesh.SimpleArrayFloat64(12)
ndarr = np.arange(12, dtype='int32')
sarr.nghost = 2

sarr[...] = ndarr[...]

print(sarr[-2]) # 0.0
print(sarr[-1]) # 1.0
print(sarr.ndarray) # [ 0.  1.  2.  3.  4.  5.  6.  7.  8.  9. 10. 11.]
print(ndarr) # [ 0  1  2  3  4  5  6  7  8  9 10 11]
```

Reply:
That is the behavior I am expecting and an expected behavior should be added in unit tests.

Reply:
resolved